### ISSUE TITLE
Sprite Flicker: Mario and Luigi Superstar Saga

### ISSUE THREAD

**Issue by LoneSolo** [mgba-emu/mgba#2623]
Sprites and objects randomly flicker while playing Mario and Luigi Superstar Saga. I can get it to occur consistently during the first cutscene, and talking to certain characters in the prologue (first character that shows you action commands, many of the characters on browsers airship). This is occurring in windows 10,11 and the steam deck.
[bug.zip](https://github.com/mgba-emu/mgba/files/9312877/bug.zip)

**Comment by LoneSolo** [comment]
I did some additional testing and found out the reason its flickering has to do with the game boy player mode. Even though  I had it turned off in settings it was still turned on through the game override modes. Its a shame it has to be turned off since this game uses rumble but it functions as expected now that the mode is turned off.

**Comment by endrift** [comment]
The game actually flickers on an real Game Boy Player. Though I'm not sure it flickers as much as it does in mGBA.

**Comment by twoco** [comment]
I can confirm on Miyoo Mini Plus. It‘s not due to transparency technique or etc. Also not on real GBA (not in this case). You can see e.g. in intro scene.  The toads with trumpets flashing / glitches. No flashing with gpSP. Not sure, but I never have seen this before. I will test this on PC soon.

**Comment by endrift** [comment]
You don't need to do additional testing in emulators. mGBA will behave the same regardless of what platform it's running on, too.

**Comment by twoco** [comment]
I tested it now and nope. On PC it works well without any issues. The sprites do not flicker on PC. 

- PC Version: 0.10.2 (Standalone); RetroArch Core 0.11-dev also no issue.
- Miyoo Mni Plus - Onion OS V4.2.0-RC - RetroArch Core 0.10-dev ec5ecb26d

Maybe it has to do with the display driver you can set on RetroArch? Miyoo Mini Plus has no screen recorder, but I could film with my smartphone. ^^ Or can we reproduce this on PC?

**Comment by endrift** [comment]
You linked to a commit that's almost a year old. That's your problem.

**Comment by twoco** [comment]
Yeah, I also suspect it. This is the version of the latest Onion OS. Not sure how to update.

**Comment by twoco** [comment]
Yeah, I also suspect it. This is the version of the latest Onion OS. Not sure how to update. But we are not 100% sure that this is the problem. And I never expected such a behavior (regardless of version age). But anyway, testing with latest core is required. I'll find out how to update and test again. Or do you have a reference to this specific fix commit / version? Can someone other post the tested version here? I mean this is always required when posting an issue! Or we could test if we can reproduce the problem with exact the same version on PC. - Update: 0.9.3 on PC also has no issues. So I would not say that the version is too old and a new would fix. But it seems that it's not easy to update the core on the device.

**Comment by twoco** [comment]
@endrift I tested the latest version 0.10.2 (Apr 24) now. Unfortunately the bug still exists. The sprites are still flickering.

**Comment by perguto** [comment]
@twoco It's not a bug, the game flickers intentionally because on the slow GBA LCD screen, this gives a transparency effect. To simulate this behavior in mgba, activate "Interframe Blending"

**Comment by endrift** [comment]
That is not correct for this game. Sprites absolutely do flicker incorrectly at times in M&L

**Comment by endrift** [comment]
That actually does happen, to some extent, on hardware if you're using a GB Player. If not, it doesn't.

**Comment by Chalking** [comment]
Unfortunately this issue still exists. Playing on the mgba RetroArch chore with Gameboy Player rumble enabled causes the sprites to flicker. Turning it off fixes the flickering. Really wish this could be fixed so I could play this game with rumble :(

**Comment by endrift** [comment]
As mentioned multiple times already, the game flickers on Game Boy Player even on real hardware.

**Comment by umjammercammy** [comment]
Hey, I'm curious about this issue because I don't fully understand- I've always played through Superstar Saga on either GB Player or DS and I couldn't recall any flickering. I'm having the flicker issue in mGBA, one of the areas in Little Fungitown is a good test because two NPCs walk on both sides of the screen and Mario and Luigi (as well as the NPCs) flicker fairly consistently when moving around.

I read this thread and dug out my GC to test and I can't reproduce this glitch on actual hardware, my mGBA save file is backed up from my real cart so I just booted up the actual cartridge with the GB Player and jumped around in that same spot repeatedly but couldn't reproduce the flickering.
